app/src/main/java/com/android/bookswap/ui/map/Map.kt

```diff
@@ -96,125 +97,125 @@ fun MapScreen(
   val longitude = geolocation.longitude.collectAsState()
   // Start location and books updates
   LaunchedEffect(Unit) {
-    bookManagerViewModel.startUpdatingBooks()
-    geolocation.startLocationUpdates()
-    cameraPositionState.position =
-        CameraPosition.fromLatLngZoom(LatLng(latitude.value, longitude.value), INIT_ZOOM)
+    if (isOnline) {
+      bookManagerViewModel.startUpdatingBooks()
+      geolocation.startLocationUpdates()
+      cameraPositionState.position =
+          CameraPosition.fromLatLngZoom(LatLng(latitude.value, longitude.value), INIT_ZOOM)
+    } else {
+      Toast.makeText(context, "Please connect to Internet to actualise", Toast.LENGTH_SHORT).show()
+    }
     isOnline = isNetworkAvailable(context)
```

Comment:
```suggestion
```
This line is a functional duplicate of §92 and serves no purpose. When the screen is first composed, all the remembered values are computed and only after that are the components and LaunchedEffects processed, so the value of `isOnline` is already computed at §92